### ISSUE TITLE
Add `ignorePath` to configuration object

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#8345]
### What is the problem you're trying to solve?

I would like to be able to do:

```js
/* stylelint.config.mjs */
export default {
  extends: 'stylelint-config-standard',
  ignorePath: ['.gitignore', '.eslintignore']
}
```

for two reasons, so that:

- https://github.com/stylelint/vscode-stylelint is aware to not lint the file when it opens it (right now it reports confusing errors)
- I can share this config with other repositories, without having to duplicate `--ignore-path`.

Ref: https://github.com/stylelint/stylelint/issues/5142#issuecomment-2564362800

### What solution would you like to see?

Let's discuss our options.

Adding it would align with [our endeavours to move more options to the configuration object](https://github.com/stylelint/stylelint/issues/5142). But we've been avoiding adding features that are opposed to a [possible flat config future](https://github.com/stylelint/stylelint/issues/7408).

ESLint deprecated their `--ignore-path` flag, and [now uses this compact method](https://eslint.org/docs/latest/use/configure/ignore#including-gitignore-files):

```js
import { includeIgnoreFile } from "@eslint/compat";
import path from "node:path";
import { fileURLToPath } from "node:url";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const gitignorePath = path.resolve(__dirname, ".gitignore");

export default [
    includeIgnoreFile(gitignorePath),
    {
        // your overrides
    }
];
```

If we did move to a flat config, we'd likely deprecate our existing ignore options like `ignorePath` and `ignoreFiles` in favour of [a single `ignores`](https://github.com/stylelint/stylelint/issues/7449) option. As our flat config future isn't set in stone, I'm not opposed to adding `ignorePath` to the config object as it's a preexisting feature. Alternatively, we can suggest the community creates a util to import ignores from files into the existing `ignoreFiles` configuration object.

**Comment by Mouvedia** [comment]
> As our flat config future isn't set in stone, I'm not opposed to adding `ignorePath` to the config object as it's a preexisting feature.

👍

> The fix could be about bringing **parity**

👍 
see also #8137

**Comment by silverwind** [comment]
Why not use existing `ignoreFiles`?

```
ignoreFiles: ['.gitignore', '.eslintignore']
```

A glob `.gitignore` must match a file name `.gitignore`, e.g. it should work exactly like eslint's [`ignores`](https://eslint.org/docs/latest/use/configure/ignore) where one can specify files or globs.

**Comment by ybiquitous** [comment]
FYI, it seems ESLint doesn't support ignoring files with the flat config. It shows an [example](https://eslint.org/docs/latest/use/configure/ignore#including-gitignore-files) of reading `.gitignore` in `eslint.config.js`.

If Stylelint adopts a flat config and simplifies ignoring paths like ESLint, the new `ignorePath` config might be unnecessary.

**Comment by silverwind** [comment]
Ah, I misunderstood. You want to load patterns from files. That seems like a possibly worthy addition, but I think by default, the list of files should probably be empty, or alternatively, it should be made possible to have "no files".

**Comment by jeddy3** [comment]
> If Stylelint adopts a flat config and simplifies ignoring paths like ESLint, the new ignorePath config might be unnecessary.

That's correct, but as our flat config future isn't set in stone and maybe some time off, adding `ignorePath` to the configuration object will bring parity between the CLI and the config object in the meantime.

I'm not against the addition; I doubt it'll add much overhead. I'll mark this as ready to implement unless there are any objections.

**Comment by ybiquitous** [comment]
Again, I tried to dig into this issue, then I got the following insights:

- There can be a conflict between the `ignorePath` and `ignoreFiles` options.
  - `ignorePath` depends on the [`ignore`](https://www.npmjs.com/package/ignore) package.
  - `ignoreFiles` depends on the [`micromatch`](https://www.npmjs.com/package/micromatch) package.
  - `ignore` is not compatible with `micromatch` (or [`minimatch`](https://www.npmjs.com/package/minimatch)).
    - See also [the `ignore` readme](https://github.com/kaelzhang/node-ignore/blob/f0bd5340a18a45433f231523c1604a36e18a5931/README.md?plain=1#L20-L24).
  - The conflict can confuse some users.
  - The conflict may cause some maintenance problems.
- To avoid these issues in the future, should Stylelint unify how to ignore files? This would take steps: deprecate and remove `ignorePath`.
- Stylelint could provide a utility function so that users could migrate from `ignorePath` to `ignoreFiles`, like [ESLint's `includeIgnoreFiles`](https://eslint.org/docs/latest/use/configure/ignore#including-gitignore-files).

If we acknowledged discussing the topic more, opening a new issue would be better than here. Personally, I think it's a good opportunity.
